lua/orgmode/org/links/handlers/file.lua

```diff
@@ -0,0 +1,152 @@
+local fs = require('orgmode.utils.fs')
+local Link = require('orgmode.org.links.link_handler')
+local Id = require('orgmode.org.links.handlers.id')
+local Internal = require('orgmode.org.links.handlers.internal')
+
+---@class OrgLinkHandlerFile:OrgLinkHandler
+---@field new fun(self: OrgLinkHandlerFile, path: string, target: OrgLinkHandlerInternal | nil, prefix: boolean | nil): OrgLinkHandlerFile
+---@field parse fun(link: string, prefix: boolean | nil): OrgLinkHandlerFile | nil
+---@field path string
+---@field skip_prefix boolean
+---@field target OrgLinkHandlerInternal | nil
+local File = Link:new('file')
+
+function File:new(path, target, skip_prefix)
+  ---@class OrgLinkHandlerFile
+  local this = Link:new()
+  this.skip_prefix = skip_prefix or false
+  this.path = path
+  this.target = target
+  setmetatable(this, self)
+  self.__index = self
+  return this
+end
+
+function File.parse(input, skip_prefix)
+  if input == nil or #input == 0 then
+    return nil
+  end
+  local deliniator_start, deliniator_stop = input:find('::')
+
+  ---@type OrgLinkHandlerInternal | nil
+  local target = nil
+  local path = input
+
+  if deliniator_start then
+    ---@class OrgLinkHandlerInternal | nil
+    target = Internal.parse(input:sub(deliniator_stop + 1), true)
+    path = input:sub(0, deliniator_start - 1)
+  end
+
+  return File:new(path, target, skip_prefix)
+end
+
+-- TODO make protocol prefix optional. Based on what?
+function File:__tostring()
+  local v = ''
+  if self.skip_prefix then
+    v = ('%s'):format(self.path)
+  else
+    v = ('%s:%s'):format(self.protocol, self.path)
+  end
+
+  if self.target then
+    v = string.format('%s::%s', v, self.target)
+  end
+
+  return v
+end
+
+function File:follow()
+  vim.cmd('edit ' .. fs.get_real_path(self.path))
+
+  if self.target then
+    self.target:follow()
+  end
+end
+
+local function autocompletions_filenames(lead)
+  local Org = require('orgmode')
```

Comment:
I see this is a very WIP, but just a note that we should not require orgmode directly anywhere. We need to pass down dependencies from top.